src/utils/EntryModel.hxx

```diff
@@ -41,389 +41,465 @@
+            {
+                value_ = valueMin_;
+            }
+            else if (value_ > valueMax_)
+            {
+                value_ = valueMax_;
+            }
         }
-        clamp();
     }
 
-    /// Get the number of significant digits
-    /// @return number of significant digits
-    unsigned digits()
+    /// Pre-increment value. While this method does prevent wrap around of
+    /// the native type limits, it is incumbent on the caller to limit the
+    /// resulting number of digits.
+    T operator ++()
     {
-        return digits_;
+        if (value_ < std::numeric_limits<T>::max())
+        {
+            ++value_;
+        }
+        clamp();
+        return value_;
     }
 
-    /// Determine if this object is holding an initial or modified value.
-    /// @return true if holding an initial value, else false if modified
-    bool has_initial()
+    /// Pre-decrement value. While this method does prevent wrap around of
+    /// the native type limits, it is incumbent on the caller to limit the
+    /// resulting number of digits.
+    T operator --()
     {
-        return hasInitial_;
+        if (value_ > std::numeric_limits<T>::lowest())
+        {
+            --value_;
+        }
+        clamp();
+        return value_;
     }
 
-    /// Clamp the value at the min or max.
-    void clamp()
+protected:
+    /// Set the min and max boundaries supported by the EntryModel parameters.
+    virtual void set_boundaries()
     {
-        if (clampCallback_)
+        valueMax_ = 0;
+        for (unsigned i = 0; i < maxSize_; ++i)
         {
-            clampCallback_();
+            valueMax_ *= base_;
+            valueMax_ += base_ - 1;
         }
+        valueMin_ = std::is_signed<T>::value ? valueMax_ / -base_ : 0;
     }
 
-private:
-    std::function<void()> clampCallback_; /**< callback to clamp value */
-    unsigned digits_     : 5; /**< number of significant digits */
-    unsigned index_      : 5; /**< present write index */
-    unsigned hasInitial_ : 1; /**< has an initial value */
-    unsigned transform_  : 1; /**< force characters to be upper case */
-    unsigned reserved_   : 20; /**< reserved bit space */
+    T value_; ///< present value held
+    T valueMin_; ///< minimum value representable by maxSize_
+    T valueMax_; ///< maximum value representable by maxSize_ 
 
-    int base_; /**< radix base */
-    char data_[N + 1]; /**< data string */
+    unsigned numLeadingZeros_  : 5; ///< number of leading zeros
+    unsigned maxSize_          : 5; ///< maximum number of digits
+    unsigned size_             : 5; ///< actual number of digits
+    unsigned isAtInitialValue_ : 1; ///< true if still has the initial value
+    unsigned empty_            : 1; ///< true if the value_ is "empty"
+    unsigned upper_            : 1; ///< force characters to be upper case
+    unsigned reserved_         : 15; ///< reserved bit space
+
+    int base_; ///< radix base
```

Comment:
It seems like this could fit into the bit field? We only need to support 10 and 16 so 5 or 6 bits should be enough.

Reply:
Fixed.